src/line_cursor.ts

```diff
@@ -475,6 +475,43 @@ export class LineCursor extends Marker {
     throw new Error('no valid nodes in this.potentialNodes');
   }
 
+  /**
+   * Get the current location of the cursor.
+   *
+   * Overrides normal Marker getCurNode to update the current node from the selected
+   * block. We typically update the node from the selection via a listener but
+   * that is not called immediately when `Gesture` calls `Blockly.common.setSelected`.
+   * In particular the listener runs after showing the context menu.
+   *
+   * @returns The current field, connection, or block the cursor is on.
+   */
+  override getCurNode(): ASTNode | null {
+    const curNode = super.getCurNode();
+    let selected = Blockly.common.getSelected();
+    if (
+      selected === null &&
+      curNode?.getType() === Blockly.ASTNode.types.BLOCK
+    ) {
+      // Selection says our curNode is not selected anymore.
+      // this.setCurNode(null as never, true);
+      return super.getCurNode();
+    }
+    if (selected?.workspace !== this.workspace) return curNode;
+    // Selection has a block in our workspace.
+    if (selected instanceof Blockly.BlockSvg) {
+      if (selected.isShadow()) {
+        // Although the shadow block is selected it's the parent that has the
+        // visual selection.
+        selected = selected.getParent();
+      }
+      if (selected) {
+        this.setCurNode(new ASTNode(ASTNode.types.BLOCK, selected), true);
```

Comment:
Could you perhaps use `Blockly.ASTNode.createBlockNode` here, instead? That's conventional with other locations in the plugin where an `ASTNode` is being created.

Reply:
Makes sense. Done as part of 1233f578fd94a0941a8e028a0417d7e83a3eda3f.